stats.go

```diff
@@ -139,7 +168,11 @@ func (t *timer) Start() Stopwatch {
 }
 
 func (t *timer) Record(interval time.Duration) {
-	t.reporter.ReportTimer(t.name, t.tags, interval)
+	if t.cachedTimer != nil {
+		t.cachedTimer.ReportTimer(interval)
+	} else {
+		t.reporter.ReportTimer(t.name, t.tags, interval)
+	}
```

Comment:
Might be worth binding this method to avoid the if/else jump at `Record` call time:

```go

type timer struct {
  // ...
  reportFn func(interval time.Duration)
}

func newTimer(/*...*/) { 
  t := &timer{/*...*/}
  if cachedTimer != nil {
    t.reportFn = t.cacheTimer.ReportTimer
  } else {
    t.reportFn = t.report
  }
}

func (t *timer) Record(interval time.Duration) {
  t.reportFn(interval)
}

func (t *timer) report(interval time.Duration) {
  t.reporter.ReportTimer(t.name, t.tags, interval)
}
```